File: app/src/main/java/br/com/vicentec12/mygames/data/source/console/ConsoleRepository.java
package br.com.vicentec12.mygames.data.source.console;

import java.util.List;

import javax.inject.Inject;
import javax.inject.Singleton;

import br.com.vicentec12.mygames.data.model.Console;
import br.com.vicentec12.mygames.data.source.Local;
import br.com.vicentec12.mygames.interfaces.Callbacks;

@Singleton
public class ConsoleRepository implements ConsoleDataSource {

    private final ConsoleDataSource consoleLocalDataSource;

    @Inject
    public ConsoleRepository(@Local ConsoleDataSource consoleLocalDataSource) {
        this.consoleLocalDataSource = consoleLocalDataSource;
    }

    @Override
    public void listConsoles(OnConsolesListedCallback callback) {
        consoleLocalDataSource.listConsoles(callback);
    }

    @Override
    public void listConsolesWithGames(OnConsolesWithGamesListedCallback callback) {
        consoleLocalDataSource.listConsolesWithGames(callback);
    }

    @Override
    public void insertConsoles(List<Console> consoles, Callbacks.OnLocalCallback callback) {
        consoleLocalDataSource.insertConsoles(consoles, callback);
    }

    @Override
    public void deleteConsole(Console console, Callbacks.OnLocalCallback callback) {
        consoleLocalDataSource.deleteConsole(console, callback);
    }

    @Override
    public void deleteAllConsoles(Callbacks.OnLocalCallback callback) {
        consoleLocalDataSource.deleteAllConsoles(callback);
    }

}

File: app/src/main/java/br/com/vicentec12/mygames/data/source/console/ConsoleRepository.kt
package br.com.vicentec12.mygames.data.source.console

import br.com.vicentec12.mygames.data.model.Console
import br.com.vicentec12.mygames.data.source.Local
import javax.inject.Inject

class ConsoleRepository @Inject constructor(
        @Local val consoleLocalDataSource: ConsoleDataSource
) : ConsoleDataSource {

    override suspend fun list() = consoleLocalDataSource.list()

    override suspend fun listWithGames() = consoleLocalDataSource.listWithGames()

    override suspend fun insert(mConsoles: List<Console>) = consoleLocalDataSource.insert(mConsoles)

    override suspend fun delete(mConsole: Console) = consoleLocalDataSource.delete(mConsole)

    override suspend fun deleteAll() = consoleLocalDataSource.deleteAll()

}